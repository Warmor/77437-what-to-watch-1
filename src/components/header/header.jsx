import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {getUser, getUserIsLoggedIn} from "~/reducer/user/selectors";

function Header(props) {
  let {
    headerTitle,
    withAuth = false,
    user,
    isLoggedIn,
  } = props;
  return <>
    <header className="page-header user-page__head">
      <div className="logo">
        <a href="main.html" className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      {headerTitle && <h1 className="page-title user-page__title">{headerTitle}</h1>}
      {withAuth && isLoggedIn && (
        <div className="user-block">
          <div className="user-block__avatar">
            <img src={user.avatar_url} alt={user.name} width="63" height="63"/>
          </div>
        </div>
      )}
      {withAuth && !isLoggedIn && (
        <div className="user-block">
          <a href="sign-in.html" className="user-block__link">Sign in</a>
        </div>
      )}
    </header>
  </>;
}

Header.propTypes = {
  headerTitle: PropTypes.string,
  withAuth: PropTypes.bool,
  user: PropTypes.object,
  isLoggedIn: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  user: getUser(state),
  isLoggedIn: getUserIsLoggedIn(state),
});

export default connect(mapStateToProps)(Header);
