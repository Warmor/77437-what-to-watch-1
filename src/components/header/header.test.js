import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import {Header} from "~/components/header/header";

Enzyme.configure({adapter: new Adapter()});

describe(`<Header/>`, () => {
  it(`Header renders correctly`, () => {
    const tree = shallow(<Header/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
