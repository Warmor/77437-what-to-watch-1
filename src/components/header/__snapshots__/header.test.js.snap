// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Header renders correctly 1`] = `
<Fragment>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <Link
        className="logo__link"
        to="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </Link>
    </div>
  </header>
</Fragment>
`;
